### PR TITLE
Few small install fixes for 1.5 and master

### PR DESCRIPTION
**Pull request by costinm**
- adjust the bootstrap for GCP
- add extra line for the error message if inject template is bad
- don't include the configmap for istiod envoy sidecar - we don't use a sidecar in this mode
- sync up the useMCP, so install with helm works
- add a comment to remember people to update global.yaml

### CODE REVIEW

File: manifests/istio-control/istio-discovery/templates/configmap-envoy.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{- if not .Values.global.istiod.enabled }}
```
Comment (howardjohn) [inline manifests/istio-control/istio-discovery/templates/configmap-envoy.yaml]: fyi, I am about to remove this in https://github.com/istio/istio/pull/20883

File: manifests/istio-control/istio-discovery/templates/deployment.yaml [hunk 2]
```diff
@@ -295,20 +295,21 @@ spec:
...
+        {{- if .Values.global.controlPlaneSecurityEnabled}}
       - name: istio-certs
         secret:
           secretName: istio.istio-pilot-service-account
           optional: true
-  {{- end }}
+        {{- end }}
+        {{- end }}
+
+      - name: config-volume
```
Comment (howardjohn) [inline manifests/istio-control/istio-discovery/templates/deployment.yaml]: what is this for? we already have the "istiod" one?

Comment (costinm) [reply manifests/istio-control/istio-discovery/templates/deployment.yaml]: No, that's the config map created by configmap.yaml, with the mesh config.

Needed in both pilot and istiod modes.

Comment (howardjohn) [reply manifests/istio-control/istio-discovery/templates/deployment.yaml]: why are we adding a volume if it isn't mounted though?

Comment (howardjohn) [reply manifests/istio-control/istio-discovery/templates/deployment.yaml]: nevermind it moved..

**Review by howardjohn — APPROVED** [review]

**Pull request merged**